applications/launchpad_v2/src/components/TitleBar/index.tsx

```diff
@@ -0,0 +1,208 @@
+import { useContext } from 'react'
+import { ThemeContext } from 'styled-components'
+import { useDispatch, useSelector } from 'react-redux'
+import { animated, useSpring } from 'react-spring'
+import { appWindow } from '@tauri-apps/api/window'
+
+import Button from '../Button'
+import Logo from '../Logo'
+import Switch from '../Inputs/Switch'
+
+import { selectExpertView } from '../../store/app/selectors'
+
+import SvgCloseCross from '../../styles/Icons/CloseCross'
+import SvgSetting from '../../styles/Icons/Setting2'
+
+import { toggleExpertView } from '../../store/app'
+import ExpertViewUtils from '../../utils/ExpertViewUtils'
+
+import {
+  LeftCol,
+  LogoContainer,
+  TitleBar as StyledTitleBar,
+  TitleBarButton,
+  WindowButtons,
+} from './styles'
+import { TitleBarProps } from './types'
+
+const TitleBar = ({ drawerViewWidth = '50%' }: TitleBarProps) => {
+  const dispatch = useDispatch()
+
+  const expertView = useSelector(selectExpertView)
+  const theme = useContext(ThemeContext)
+
+  const [expertViewSize] = ExpertViewUtils.convertExpertViewModeToValue(
+    expertView,
+    drawerViewWidth,
+  )
+
+  const drawerContainerStyle = ExpertViewUtils.drawerAnim(expertViewSize)
+
+  const logoColorAnim = useSpring({
+    color: expertView === 'fullscreen' ? theme.background : theme.primary,
+  })
+
+  const buttonIconStyle = { width: 12, height: 12 }
+
+  const titleBarBgAnim = useSpring({
+    background: theme.backgroundSecondary,
+  })
+
+  const onMinimize = () => {
+    appWindow.minimize()
+  }
+
+  const onMaximize = () => {
+    appWindow.maximize()
+  }
+
+  const onClose = () => {
+    appWindow.close()
+  }
+
+  const onExpertViewClick = () => {
+    if (expertView !== 'hidden') {
+      dispatch(toggleExpertView('hidden'))
```

Comment:
why use string for a boolean open/closed state?

Reply:
I saw below that there's a third 'fullscreen' option, in which case 'toggle' is a misleading name, imo

either "setExpertView" or store fullscreen as one boolean, open as another (then you can toggle both)

Reply:
good catch, `setExpertView` fits better than `toggle...`